wrapper_writer/scala_parser.py

```diff
@@ -0,0 +1,173 @@
+        return_type = last
+        return return_type
+
+    @staticmethod
+    def extract_method_name(raw_res):
+        """
+        The function will use regex to extract the method name from the logic code
+        :param raw_res: A method signature from the file to be parsed
+        :return: clean_func_name: String object of the function name
+        """
+        func_name = r"def (\w+)"
+        func_name_find = re.search(func_name, raw_res)
+        fun_name_raw = func_name_find.group()
+        clean_func_name = fun_name_raw.replace("def ", "")
+        return clean_func_name
+
+    @staticmethod
+    def extract_params(raw_res):
+        """
+        This function will use regex to extract the parameters from the logic code
+        :param raw_res: method signature from the file to be parsed
+        :return new_dict: A dictionary containing the parameters
+        """
+        retrieve_params = r"\((.*)\)"
+        retrieve_params_find = re.search(retrieve_params, raw_res)
+        retrieve_params_raw = retrieve_params_find.group()
+        retrieve_params_clean = retrieve_params_raw[1:-1]
+        dict_by_comma = dict(item.split(":") for item in retrieve_params_clean.split(","))
+        for k, v in dict_by_comma.items():
+            dict_by_comma[k] = v.lstrip()
+            new_dict = {k.lstrip(): v for k, v in dict_by_comma.items()}
+        return new_dict
+
+    def multi_process(self):
+        """
+        This function will process each method found and write it to a yaml file
+        :return:
+        """
+        all_found = self.find_method_regex()
+        matches = tuple(all_found)
+        if not matches:
+            raise Exception("No Methods Found")
+        for i in matches:
+            ig = i.group()
+            base_raw = os.path.basename(self.filename)
+            container_name = os.path.splitext(base_raw)[0]
+            return_type = self.extract_return_type(ig)
+            method_name = self.extract_method_name(ig)
+            params = self.extract_params(ig)
+            data = {container_name: {method_name: {"params": params, "returns": return_type}}}
+            with open(self.config_filename, 'a') as yaml_file:
+                yaml.dump(data, yaml_file, default_flow_style=False)
+
+
+class App:
+    """
+    This App class orchestrates the scala parser application
+    :param folder: String object of the folder path which is provided by the user to indicate where files live
+    :param file: String object if the file to be parsed
+    :param file_extension: String object of file extension to look for
+    """
+
+    def __init__(self, folder=None, logic_file=None, file_extension="*.scala", config_name="config_sadhg.yml", append_config=False):
+        self.folder = folder
+        self.file = logic_file
+        self.file_extension = file_extension
+        self.config_file = config_name
+        self.append_config = append_config
+
+    def delete_config(self):
```

Comment:
I dont really have a problem with having something like this but the design will need to be updated and I'm not sure what the use case for this is